pkg/adminconfig/rego_file_evaluator_test.go

```diff
@@ -74,13 +78,134 @@ func NewEvaluator() *adminconfig.RegoPolicyEvaluator {
 	return &adminconfig.RegoPolicyEvaluator{Log: logging.LogInit("test", "ConfigPolicyEvaluator"), ReadyForEval: true, Query: query}
 }
 
+func EvaluatorWithInfrastructure() *adminconfig.RegoPolicyEvaluator {
+	module := `
+		package adminconfig
+
+		# no copy for dev workloads
+		config[{"copy": decision}] {
+			input.request.usage.read == true
+			input.workload.properties.stage == "DEV"
+			policy := {"description": "do not copy in DEV workload"}
+			decision := {"policy": policy, "deploy": false}
+		}
+
+		# Cost Efficient Production Workloads - read
+		config[{"read": decision}] {
+			input.request.usage.read == true
+			input.workload.properties.stage == "PROD"
+			input.workload.properties.priority != "high"
+			dataset_region := input.request.dataset.geography
+			workload_region := input.workload.cluster.metadata.region			
+			data.infrastructure.bandwidth.values[dataset_region][workload_region] == "S"
+			policy := {"description": "use cheaper storage"}
+			clusters := { "metadata.region" : [ workload_region ] }
+			decision := {"policy": policy, "deploy": true, "restrictions": {"clusters": clusters}}
+		}
+
+		# Cost Efficient Production Workloads - copy
+		config[{"copy": decision}] {
+			input.request.usage.read == true
+			input.workload.properties.stage == "PROD"
+			input.workload.properties.priority != "high"
+			dataset_region := input.request.dataset.geography
+			workload_region := input.workload.cluster.metadata.region			
+			data.infrastructure.bandwidth.values[dataset_region][workload_region] == "S"
+			policy := {"description": "use cheaper storage"}
+			accounts := [ data.infrastructure.storageaccounts.values[i].id | data.infrastructure.storageaccounts.values[i].cost <= "80"; 
+																	  		 data.infrastructure.storageaccounts.values[i].type == "object-storage";
+																	         data.infrastructure.bandwidth.values[data.infrastructure.storageaccounts.values[i].region][workload_region] != "S" ]
+			decision := {"policy": policy, "deploy": true, "restrictions": {"storageaccounts": {"values.id": accounts}}}
```

Comment:
"id" instead of "values.id" - the filtering will be done for FybrikStorageAccounts according to Spec.id

Reply:
Changed